lib/Dialect/AIEVec/IR/AIEVecOps.cpp

```diff
@@ -389,30 +394,40 @@ void aievec::FMAOp::print(OpAsmPrinter &p) {
 // Verify Mul and FMA op.
 template <typename T> LogicalResult verifyMulFMAOp(T op) {
   // Verify the types
-  aievec::AccType resultType =
-      op.result().getType().template dyn_cast<aievec::AccType>();
   VectorType lhsType = op.lhs().getType().template dyn_cast<VectorType>();
   VectorType rhsType = op.rhs().getType().template dyn_cast<VectorType>();
 
   if (!lhsType || !rhsType)
     return op.emitError("requires vector type");
-  if (!resultType)
+
+  // Determine if it is an integer or float operation
+  bool isInt = lhsType.getElementType().isa<IntegerType>();
```

Comment:
Perhaps it would make sense to also separate the integer and floating point cases into separate ops?

Reply:
This is something that I was contemplating as well. However, with the upcoming PR for #101, I end up not actually needing to differentiate between integer and float all that much since I mostly did it for the sake of dealing with AccType, the exception being to determine whether a UPS is needed when the AccType isn't around to serve as the trigger. This does seem like something to investigate as part of figuring out how abstract we want the dialect to be.

Reply:
Good point, forgot there was a separate issue for it.